tests/integration/test_hive.py

```diff
@@ -0,0 +1,409 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+# pylint:disable=redefined-outer-name
+
+import math
+import uuid
+from typing import Dict
+from urllib.parse import urlparse
+
+import pyarrow.parquet as pq
+import pytest
+from pyarrow.fs import S3FileSystem
+
+from pyiceberg.catalog import Catalog, load_catalog
+from pyiceberg.exceptions import NoSuchTableError
+from pyiceberg.expressions import (
+    And,
+    EqualTo,
+    GreaterThanOrEqual,
+    IsNaN,
+    LessThan,
+    NotEqualTo,
+    NotNaN,
+)
+from pyiceberg.io.pyarrow import pyarrow_to_schema
+from pyiceberg.schema import Schema
+from pyiceberg.table import Table
+from pyiceberg.types import (
+    BooleanType,
+    IntegerType,
+    NestedField,
+    StringType,
+    TimestampType,
+)
+
+DEFAULT_PROPERTIES: Dict[str, str] = {}
+
+
+@pytest.fixture()
+def catalog() -> Catalog:
+    return load_catalog(
+        "local",
+        **{
+            "type": "hive",
+            "uri": "http://localhost:9083",
+            "s3.endpoint": "http://localhost:9000",
+            "s3.access-key-id": "admin",
+            "s3.secret-access-key": "password",
+        },
+    )
+
+
+@pytest.fixture()
+def table_test_null_nan(catalog: Catalog) -> Table:
+    return catalog.load_table("default.test_null_nan")
+
+
+@pytest.fixture()
+def table_test_null_nan_rewritten(catalog: Catalog) -> Table:
+    return catalog.load_table("default.test_null_nan_rewritten")
+
+
+@pytest.fixture()
+def table_test_limit(catalog: Catalog) -> Table:
```

Comment:
I wonder if we could use `pytest_lazy_fixture` to avoid duplicate tests in `test_rest.py` and `test_hive.py`
like what we did in https://github.com/apache/iceberg-python/blob/2bd8cf21fe92edc0cca2fc9b39d676d48abfafec/tests/catalog/test_sql.py#L113-L120.

Here it can be
```python
@pytest.fixture(params=[
    pytest.lazy_fixture('catalog_hive'),
    pytest.lazy_fixture('catalog_rest')
])
def catalog(request):
    return request.param

@pytest.fixture()
def table_test_limit(catalog: Catalog) -> Table:
```
and `table_test_limit` should load table from both catalogs in turn

I modified an example from https://pypi.org/project/pytest-lazy-fixture/
```python
@pytest.fixture(params=[
    pytest.lazy_fixture('one'),
    pytest.lazy_fixture('two')
])
def combined(request):
    return request.param

@pytest.fixture
def one():
    return 1

@pytest.fixture
def two():
    return 2

@pytest.fixture
def combined2(combined):
    return combined

def test_func(combined2):
    assert combined2 in [1, 2]
```
which seems to work.

I will try to test this in our integration test later. If it works, I think it can be worth to implement. What do you think?

Reply:
I added the lazy fixtures in https://github.com/apache/iceberg-python/pull/178 and I waited for the PR to be merged because I wanted to add them here as well. The reason I didn't do it is because `test_upgrade_table_version` and `test_table_properties` are not implemented in Hive (yet). This would mean that we need to keep them in parity which might slow down development.

Reply:
Thanks for the context! For `test_upgrade_table_version` and `test_table_properties`, I think we might skip them in test_hive by
```python
@pytest.fixture()
def table(catalog: Catalog) -> Table:
  if catalog.name == "hive":
    pytest.skip("Not Implemented: ...")
    ...

@pytest.fixture()
def table_test_table_version(catalog: Catalog) -> Table:
    if catalog.name == "hive":
      pytest.skip("Not Implemented: ...")
     ...
```
(We might also find a way to put `skip` into the decorator)
Since the two fixtures are solely used by the two tests respectively.

I think this can be discussed further later. The current PR has already done many things and it is important to updating table tests

Reply:
I actually like this suggestion a lot, let me add this to the PR